lib/r10k/action/deploy/module.rb

```diff
@@ -1,30 +1,15 @@
-require 'r10k/util/setopts'
 require 'r10k/deployment'
 require 'r10k/action/visitor'
-require 'r10k/logging'
+require 'r10k/action/base'
 require 'r10k/deployment/write_lock'
 
 module R10K
   module Action
     module Deploy
-      class Module
+      class Module < R10K::Action::Base
 
-        include R10K::Logging
-        include R10K::Util::Setopts
         include R10K::Deployment::WriteLock
 
-        def initialize(opts, argv)
-          @opts = opts
-          @argv = argv
-          setopts(opts, {
-            :config      => :self,
-            :environment => nil,
-            :trace       => :self
-          })
-
-          @purge = true
```

Comment:
`:config` and `:trace` are handled by the parent `#initialize`, which we can rely on directly.